lib/mod/cache/lookup.go

```diff
@@ -9,7 +9,7 @@ import (
 )
 
 func Load(lang, mod, ver string) (FS billy.Filesystem, err error) {
-	flds := strings.Split(mod, "/")
+	flds := strings.SplitN(mod, "/", 3)
```

Comment:
not sure how this works for monorepos, which may have more than 3 parts. I'm inclined to leave it the way it was and check if the length of `flds` equals 3 and print an error that this is not currently supported.

Also not sure if it would work for CUE repo either, which only has two components

Reply:
Ah, I see this may be for the GitLab infamy? We should probably add a check for gitlab and try to fetch the archive file if that is your primary use case.

Reply:
Yes indeed it's for the GitLab infamy... You mean by using the GitLab api?

Reply:
If doing something custom for GitLab, I guess we'll have to try every path to distinguish repo path from monorepo internal directory, right? Maybe we could use the same thing `go` is doing (enforcing the `.git` suffix).

Reply:
is the last component always the repo?

Will have to look into how Go handles these things